storagenode/storagenodedb/database.go

```diff
@@ -56,27 +101,38 @@ func New(log *zap.Logger, config Config) (*DB, error) {
 	}
 	pieces := filestore.New(piecesDir)
 
-	infodb, err := newInfo(config.Info2)
-	if err != nil {
-		return nil, err
-	}
-
 	dbs, err := boltdb.NewShared(config.Kademlia, kademlia.KademliaBucket, kademlia.NodeBucket, kademlia.AntechamberBucket)
 	if err != nil {
 		return nil, err
 	}
 
-	return &DB{
+	db := &DB{
 		log: log,
 
 		pieces: pieces,
 
-		info: infodb,
-
 		kdb: dbs[0],
 		ndb: dbs[1],
 		adb: dbs[2],
-	}, nil
+	}
+
+	versionsPath := config.Info2
+	versionsDb, err := openDatabase(versionsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Initialize databases. Currently shares one info.db database file but
+	// in the future these will initialize their own database connections.
+	db.versions = newVersions(versionsDb, versionsPath)
+	db.pieceinfo = newPieceInfo(versionsDb, versionsPath)
+	db.vouchers = newVouchers(versionsDb, versionsPath)
+	db.usedserials = newUsedSerials(versionsDb, versionsPath)
+	db.bandwidth = newBandwidth(versionsDb, versionsPath)
+	db.console = newConsole(versionsDb)
+	db.orders = newOrders(versionsDb, versionsPath)
```

Comment:
Why not define these above where we define the struct literal?

Reply:
Done!